docs/DISTRIBUTION.md

```diff
@@ -128,12 +128,50 @@ Each run creates all hourly and daily archives for the last 14 days. To prevent
+
+- export.bin - the binary containing the exposure keys
+- export.sig - the signature to verify the export binary
+
+The `export.bin` file of the archive contains the Temporary Exposure Keys that are broadcasted by the devices of the
+people who are diagnosed with COVID-19.
+The `export.sig` file is used by the server to sign each `export.bin` with a private key. The `export.sig` file
+contains the raw signature as well as additional information needed for verification.
 
 ## Signing
 
-Coming soon
+The signing of the files basically means creating and signing an archive that contain the `export.bin` file
+(the file name is configurable in `application.yaml` under `tek-export.file-name` property).
+Both the application configuration as well as the Temporary Exposure Keys are signed so that the mobile devices
+can verify if the files are originating from the right backend. Using the signature, attack vectors like a
+man in the middle attack, request forgery etc. can be mitigated successfully. The implementation is done in
+[`AppConfigurationSigningDecorator`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/structure/archive/decorator/signing/AppConfigurationSigningDecorator.java).
+
+The signing process takes the `export.bin` archive and creates a signature of the data structure which is then written into the `export.sig`.
+Please see`getSignatureFile` method located in [`SigningDecoratorOnDisk`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/structure/archive/decorator/signing/SigningDecoratorOnDisk.java).
+There are different implementations for that, i.e. how to sign an app configuration which is the
+`AppConfigurationSigningDecorator` - the specific code to determine how an application configuration file is supposed to be signed.
+For the Temporary Exposure Key files, they are signed in the similar way using the
+[`DiagnosisKeySigningDecorator`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/archive/decorator/signing/DiagnosisKeySigningDecorator.java).
+
+The algorithm used for signing the archives as well as other relevant information is defined in the [`application.yaml`](/services/distribution/src/main/resources/application.yaml).
 
 ## Bundling and Shifting
 
-Coming soon
+The [`DiagnosisKeyBundler`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/DiagnosisKeyBundler.java)
+is responsible for the core logic of the distribution service, e.g. how the diagnosis keys are grouped into which date and which hour folder
+and necessary transformations.
+
+In the preparation step, the distribution job queries all diagnosis keys that are available in the database and
+cache them into the `DiagnosisKeyBundler`. The purpose of this class is to distribute keys into different hours and dates and tell
```

Comment:
"The purpose of this class is to arrange keys based on their earliest time (date and hour) of distribution, and provide this information to the components which create the archives."